src/palladio/AnnotationParsing.h

```diff
@@ -32,23 +35,14 @@ struct EnumAnnotation {
 	std::vector<std::wstring> mOptions;
 };
 
-using RangeAnnotation = std::pair<double, double>;
+struct RangeAnnotation {
+	std::pair<double, double> minMax;
+	bool restricted;
+};
 using FileAnnotation = std::vector<std::wstring>;
 using ColorAnnotation = std::array<double, 3>;
 
-struct AttributeAnnotationInfo {
-	const prt::Annotation& mAnnotation;
-	AttributeTrait mAttributeTrait;
-	std::wstring mDescription;
-
-	AttributeAnnotationInfo(const prt::Annotation& annotation, AttributeTrait attributeTrait, const std::wstring& description)
-	    : mAnnotation(annotation), mAttributeTrait(attributeTrait), mDescription(description) {}
-	AttributeAnnotationInfo() = delete;
-	AttributeAnnotationInfo(const AttributeAnnotationInfo&) = delete;
-	AttributeAnnotationInfo(AttributeAnnotationInfo&&) = delete;
-	AttributeAnnotationInfo& operator=(const AttributeAnnotationInfo&) = delete;
-	AttributeAnnotationInfo& operator=(AttributeAnnotationInfo&&) = delete;
-};
+using AnnotationVariant = std::variant<std::monostate, EnumAnnotation, RangeAnnotation, FileAnnotation, std::wstring>;
```

Comment:
This name is not very intuitive to me. How about `AnnotationTraitParameter` or `AnnotationParameter`?

Similarly, `AttributeTrait` above is too generic to me, how about `AnnotationTrait`?

Reply:
renamed to AnnotationTraitParameter